File: CS/10_StampsAndWatermarks/FillStrokeText.cs
using System;
using System.Drawing;
using System.Windows.Forms;
using Spire.Pdf;
using Spire.Pdf.Graphics;
using Spire.Pdf.Tables;

namespace FillStrokeText
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            // Create a new PDF document and load the file from disk
            PdfDocument doc = new PdfDocument();
            doc.LoadFromFile(@"../../../../../../Data/PDFTemplate_N.pdf");

            // Get the first page of the document
            PdfPageBase page = doc.Pages[0];

            // Define a PDF pen with gray color
            PdfPen pen = new PdfPen(Color.Gray);

            // Save the current graphics state
            PdfGraphicsState state = page.Canvas.Save();

            // Rotate the page canvas by -20 degrees
            page.Canvas.RotateTransform(-20);

            // Create a PdfStringFormat object and set the character spacing to 5f
            PdfStringFormat format = new PdfStringFormat();
            format.CharacterSpacing = 5f;

            // Draw the string "E-ICEBLUE" on the page using a specified font, pen, position, and format
            page.Canvas.DrawString("E-ICEBLUE", new PdfFont(PdfFontFamily.Helvetica, 45f), pen, 0, 500f, format);

            // Restore the graphics state to its previous state
            page.Canvas.Restore(state);

            // Save the modified PDF document to the specified output file path
            string output = "FillStrokeText_out.pdf";
            doc.SaveToFile(output);

            // Close the PDF document
            doc.Close();

            //Launch the Pdf file
            PDFDocumentViewer(output);
        }

        private void PDFDocumentViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

    }
}

File: VB/10_StampsAndWatermarks/FillStrokeText.vb
Imports Spire.Pdf
Imports Spire.Pdf.Graphics

Namespace FillStrokeText
    Partial Public Class Form1
        Inherits Form
        Public Sub New()
            InitializeComponent()
        End Sub
        Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
            ' Create a new PdfDocument object
            Dim doc As New PdfDocument()

            ' Load the PDF document from the specified file path
            doc.LoadFromFile("../../../../../../Data/PDFTemplate_N.pdf")

            ' Get the first page of the document
            Dim page As PdfPageBase = doc.Pages(0)

            ' Create a new PdfPen object with gray color
            Dim pen As New PdfPen(Color.Gray)

            ' Save the current graphics state of the page's canvas
            Dim state As PdfGraphicsState = page.Canvas.Save()

            ' Rotate the canvas by -20 degrees
            page.Canvas.RotateTransform(-20)

            ' Create a new PdfStringFormat object
            Dim format As New PdfStringFormat()

            ' Set the character spacing to 5
            format.CharacterSpacing = 5.0F

            ' Draw a string "E-ICEBLUE" on the canvas using specified font, pen, position, and format
            page.Canvas.DrawString("E-ICEBLUE", New PdfFont(PdfFontFamily.Helvetica, 45.0F), pen, 0, 500.0F, format)

            ' Restore the graphics state of the canvas to its previous state
            page.Canvas.Restore(state)

            ' Define the output file path
            Dim output As String = "FillStrokeText_out.pdf"

            ' Save the modified document to the output file
            doc.SaveToFile(output)

            ' Close the document
            doc.Close()

            ' Launch the Pdf file
            PDFDocumentViewer(output)
        End Sub

        Private Sub PDFDocumentViewer(ByVal fileName As String)
            Try
                Process.Start(fileName)
            Catch
            End Try
        End Sub

    End Class
End Namespace
